src/compute-engine/library/arithmetic.ts

```diff
@@ -641,7 +641,6 @@ export const ARITHMETIC_LIBRARY: IdentifierDefinitions[] = [
           });
           return sumNeg % 2 === 0 ? 'non-positive' : 'non-negative';
         }
-        if (ops.every((x) => !x.is(0) && x.isReal)) return 'real-not-zero';
         if (ops.every((x) => !x.is(0))) return 'not-zero';
```

Comment:
There is a check for is(0) line 619, so I don't think this is ever reached. It also returns 'not-zero', while the check line 619 returns other values. Not sure which is correct.

Reply:
This is a problem with .is(0). It should return true, false, or undefined. If you do that (which I think I assumed it did) then !x.is(0) and x.is(0) can both be false. Otherwise, x^2(2) will be considered not 0 since (x^2).is(0) returns false.

Reply:
`x.is(0)` is either true or false. Something either *is* 0, or it is not. Perhaps you meant to check that `x.sgn` was `zero` or not?

Reply:
I see. Those checks should be using x.sgn not is(0).

Reply:
is(0) works for showing something is zero but !is(0) cannot be used to check if something is non-zero (only if it is not identically 0).